ArduPlane/quadplane.cpp

```diff
@@ -2577,8 +2661,20 @@ bool QuadPlane::verify_vtol_land(void)
     if (!available()) {
         return true;
     }
-    if (poscontrol.state == QPOS_POSITION2 &&
-        plane.auto_state.wp_distance < 2) {
+
+#if PRECISION_LANDING == ENABLED
+    AC_PrecLand &precland = plane.precland;
+    const bool navigating = pos_control->is_active_xy();
+    bool doing_precision_landing = precland.target_acquired() && navigating;
+    Vector2f target;
+    plane.precland.get_target_position_relative_cm(target);
+#else
+    bool doing_precision_landing = false;
+#endif
+
+    if ((poscontrol.state == QPOS_POSITION2 && plane.auto_state.wp_distance < 2) ||
+        (doing_precision_landing && target.length() < 2)) {
```

Comment:
There is similar code in control_loiter (lines 1231-1257) which was previously written that also controls landing state transitions. Wasn't sure to put it there or here - where there is a QPOS_LAND_DESCEND transition already defined.